hazelcast/src/main/resources/release_notes.txt

```diff
@@ -1,5 +1,29 @@
 This document lists the new features, enhancements, fixed issues and, removed or deprecated features for Hazelcast IMDG 3.12.x release. The numbers in the square brackets refer to the issues in Hazelcast's GitHub repositories.
 
+==== 3.12.3 ====
+
+1. Enhancements
+
+- Added a new implementation of SecondsBasedEntryTaskScheduler for the FOR_EACH mode to improve the performance of TransactionContext.commit(). [#15586]
+- Improved the client graceful shutdown mechanism so that its tasks are executed before marking a client as inactive. [#15565]
+- Added sent/received bytes logs for each ProtocolType into the diagnostics. [#15541]
+- Improved the performance of TransactionLog.add() by avoiding the LinkedList.remove() call. [#15435]
+- To be shown on Management Center, the clients now send both its IP address and canonical hostname. Before, only the hostname of the client was shown. [#15421]
+- Added the support for nested JSON objects in the arrays. [#15368]
+
+2. Fixes
+
+- Fixed an issue where the node.isMaster() method could cause inconsistent behaviors when it is executed during a partition service lock. [#15618]
+- Fixed an issue where Management Center was not able to see the 3.12.2 members (members with CP subsystem enabled) after performing a rolling upgrade from Hazelcast IMDG 3.11.4 to 3.12.2. 'ManagementCenterService' was shutting down and throwing an exception when a CP-subsystem-enabled member was trying to report to Management Center for the first time. [#15609]
```

Comment:
One minor thing: https://github.com/hazelcast/hazelcast/pull/15609 is not for 3.12.3 but https://github.com/hazelcast/hazelcast/pull/15621 is.